django/db/models/query.py

```diff
@@ -583,23 +583,18 @@ def update_or_create(self, defaults=None, **kwargs):
         specifying whether an object was created.
         """
         defaults = defaults or {}
-        self._for_write = True
```

Comment:
This line is necessary because we can open transaction on a wrong database (read vs. write).

Reply:
It is required **before** `transaction.atomic()` call, right? I'm asking because there is also `self._for_write = True` in  `get_or_create` method.

Reply:
Ah, ok, I found that is should be before getting `self.db` actually.

Reply:
Yes.